sympy/solvers/solveset.py

```diff
@@ -978,6 +989,349 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _expo_solver(f):
+    r"""
+    Helper function for solving exponential equations.
+
+    Exponential equations are the type of equations which includes a
+    variable located in the exponent.
+    For example
+
+    .. math:: 5^{(2*x + 3)} - 5^{(3*x - 1)}
+
+    The function evaluates exponetial equations having two arguments,
+    i.e, exponential equations having only two exponent terms
+    (like the one above).
+
+    The helper takes the equation as the input, tries to reduce the
+    equation to log form (if possible) and returns the modified
+    equation.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _expo_solver as expo_solver
+    >>> from sympy import symbols
+    >>> x = symbols('x', real=True)
+    >>> expo_solver(3**(2*x) - 2**(x + 3))
+    2*x*log(3) - (x + 3)*log(2)
+
+
+    * Proof of correctness of the method
+
+    The logarithm is the inverse function to exponentiation.
+    The defining relation between exponentiation and logarithm is:
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    Therefore if we are given an equation with exponent terms, we can
+    convert every term to its corresponding log form. This is achieved by
+    taking logarithms and expanding them using log identities so that the
+    equations can be easily handled by `solveset`.
+
+    For example:
+
+    .. math:: 3^{2*x} = 2^{(x + 3)}
+
+    Taking log both sides will reduce the equation to
+
+    .. math:: (2*x)*log(3) = (x + 3)*log(2)
+
+    This form can be easily handed by `solveset`.
+    """
+
+    try:
+        arg1, arg2 = ordered(f.args)
+    except TypeError:
+        raise ValueError("Equations with more than two arguments are not\
+            supported")
+
+    lhs = arg1
+    rhs = -arg2
+
+    lhs = expand_log(log(lhs))
+    rhs = expand_log(log(rhs))
+
+    return lhs - rhs
+
+
+def _check_expo(f, symbol):
+    r"""
+    Helper to check whether an equation is exponential or not.
```

Comment:
This line just says that it checks if its exponential or not but the description below says it checks only two exponent terms, this is not consistent and misleading.

Reply:
Also why does it only looks for exponential with two exponent terms? In `_transolve`'s documentation you have written it can solve Exponential equations which is not correct, based on this.

Reply:
> Also why does it only looks for exponential with two exponent terms? 

Exponents with one term is handled by `_invert_real`, more than two terms are not supported (unless they can be reduced like `4**(x + 1) + 4**(x + 2) + 4**(x - 1) - 3**(x + 2) - 3**(x + 3)`, this will be factored and handled in `is_Mul` in `_transolve`), therfore `_check_expo` only looks for two terms.
Should I make for n terms??

Reply:
Which one do you think is better and why and how would you convey this information to a developer? Think about these things.

Reply:
I have made changes in `_check_expo` to work for n arguments.